react-toggle-switch-component/src/toggle-switch.jsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+export default class Toggle extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { on: false };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    if (this.state.on) {
+      this.setState({ on: false });
+    } else {
+      this.setState({ on: true });
+    }
+  }
+
+  render() {
+    if (this.state.on) {
+      return <div onClick={this.handleClick} className="on">
+        <button> </button>
+        <h1>On</h1>
+        </div>;
+    } else {
+      return <div onClick={this.handleClick} className="off">
+        <button> </button>
+        <h1>Off</h1>
+      </div>;
+    }
```

Comment:
To avoid duplicating your JSX structure it would be best to use the `if` statement to set a couple variables that would set the pieces that change. For example:

```JSX
let switchClass = 'off';
let switchText = 'Off';

if (this.state.on) {
  switchClass = 'on';
  switchText = 'On';
}

return (
  <div onClick={this.handleClick} className={switchClass}>
    <button> </button>
    <h1>{switchText}</h1>
  </div>
);
```